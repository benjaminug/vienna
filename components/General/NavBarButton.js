import React, { Component, PropTypes } from 'react'
import { Text, TouchableOpacity, View, StyleSheet } from 'react-native'
import Icon from 'react-native-vector-icons/FontAwesome'

export default class NavbarButton extends Component {

	static propTypes = {
		children: PropTypes.string,
		onPress: PropTypes.func,
	}

	static defaultProps = {
		onPress: () => ({}),
	}

	render() {
		const { tintColor, margin, onPress } = this.props;
		return (
			<TouchableOpacity onPress={onPress}>
				<View style={styles.container}>
					<Icon size={22} name="chevron-left" color="#333333" />
					<Text numberOfLines={1} style={[styles.text]}>{this.props.children}</Text>
				</View>
			</TouchableOpacity>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		flexDirection: 'row',
		paddingLeft: 10,
	},
	text: {
		color: '#333333',
		lineHeight: 18,
		marginLeft: 5,
		width: 80,
	}
})
