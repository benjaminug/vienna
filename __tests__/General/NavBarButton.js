import 'react-native'
import React from 'react'
import NavBarButton from '../../components/General/NavBarButton'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
	const tree = renderer.create(
		<NavBarButton />
	).toJSON()
	expect(tree).toMatchSnapshot()

	const withText = renderer.create(
		<NavBarButton>Back</NavBarButton>
	).toJSON()
	expect(withText).toMatchSnapshot()
});
