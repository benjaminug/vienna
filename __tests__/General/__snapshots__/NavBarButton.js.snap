exports[`test renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function bound touchableHandleResponderGrant]}
  onResponderMove={[Function bound touchableHandleResponderMove]}
  onResponderRelease={[Function bound touchableHandleResponderRelease]}
  onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
  onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
  onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
  style={
    Object {
      "opacity": 1
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingLeft": 10
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#333333",
            "fontFamily": "FontAwesome",
            "fontSize": 22,
            "fontStyle": "normal",
            "fontWeight": "normal"
          },
          undefined
        ]
      }>
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#333333",
            "lineHeight": 18,
            "marginLeft": 5,
            "width": 80
          }
        ]
      } />
  </View>
</View>
`;

exports[`test renders correctly 2`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function bound touchableHandleResponderGrant]}
  onResponderMove={[Function bound touchableHandleResponderMove]}
  onResponderRelease={[Function bound touchableHandleResponderRelease]}
  onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
  onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
  onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
  style={
    Object {
      "opacity": 1
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingLeft": 10
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#333333",
            "fontFamily": "FontAwesome",
            "fontSize": 22,
            "fontStyle": "normal",
            "fontWeight": "normal"
          },
          undefined
        ]
      }>
      
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#333333",
            "lineHeight": 18,
            "marginLeft": 5,
            "width": 80
          }
        ]
      }>
      Back
    </Text>
  </View>
</View>
`;
